src/index.js

```diff
@@ -110,9 +116,14 @@ const mapToPropertyDescriptor = (
         : value;
     },
     set(value) {
+      if (typeOrSerDes === Types.json && typeof value === 'boolean') {
+        setBooleanAttribute(this, name, value);
+        return;
+      }
+
       const attributeValue = (() => {
         if (typeOrSerDes === Types.json) {
-          return JSON.stringify(value);
+          return typeof value === 'string' ? value : JSON.stringify(value);
```

Comment:
Won't `typeof value === 'string'` always be true? How do you want to differenciate a JSON string and a non JSON string?